js/algoliasearch/internals/frontend/common.js

```diff
@@ -143,17 +142,33 @@ document.addEventListener("DOMContentLoaded", function (e) {
 				hit.price = hit.price[0];
 			}
 
-			if (hit['price'] !== undefined && price_key !== '.' + algoliaConfig.currencyCode + '.default' && hit['price'][algoliaConfig.currencyCode][price_key.substr(1) + '_formated'] !== hit['price'][algoliaConfig.currencyCode]['default_formated']) {
-				hit['price'][algoliaConfig.currencyCode][price_key.substr(1) + '_original_formated'] = hit['price'][algoliaConfig.currencyCode]['default_formated'];
+			if (Array.isArray(hit.price_with_tax)) {
+				hit.price_with_tax = hit.price_with_tax[0];
 			}
-			
-			if (hit['price'] !== undefined && hit['price'][algoliaConfig.currencyCode]['default_original_formated']
-				&& hit['price'][algoliaConfig.currencyCode]['special_to_date']) {
+
+			// price_key is in format .CURRENCY.KEY, fix that for array access
+			const price_key_as_key = price_key.replace('.' + algoliaConfig.currencyCode + '.', '');
```

Comment:
From here I re-wrote the final section, as the Product Entity Helper is now putting out 4 fields per group (group_x, x_formated, x_original, x_original_formated) I think I "simplified" this js by checking if the date is in the range for the special_price and if it is not then replacing x_formated with x_original_formated